test/functional/vimscript/system_spec.lua

```diff
@@ -190,6 +190,27 @@ describe('system()', function()
       eq(iswin() and '??\n' or 'ああ\n', eval([[system('Write-Output "ああ"')]]))
     end)
 
+    it('check powershell filter and redirection (#16271)', function()
+      if iswin() then
+        local tempfile = helpers.tmpname()
+        nvim('set_option', 'shell', 'pwsh')
+        nvim('set_option', 'shellcmdflag', '-NoLogo -NoProfile -ExecutionPolicy RemoteSigned -Command [Console]::InputEncoding=[Console]::OutputEncoding=[System.Text.Encoding]::UTF8;')
+        nvim('set_option', 'shellredir', '-RedirectStandardOutput %s -NoNewWindow -Wait')
+        nvim('set_option', 'shellpipe', '-RedirectStandardOutput %s -NoNewWindow -Wait')
+        nvim('set_option', 'shellquote', '')
+        nvim('set_option', 'shellxquote', '')
```

Comment:
can use `helpers.set_shell_powershell()` instead (which also needs to be updated--all other powershell tests will use it)